packages/commonwealth/server/passport/magic.ts

```diff
@@ -207,6 +206,59 @@ async function createNewMagicUser({
   });
 }
 
+// Replaces ghost addresses once replacements have been generated
+async function replaceGhostAddresses(
+  existingUserInstance: UserInstance,
+  addressInstances: AddressInstance[],
+  models: DB,
+  transaction: Transaction,
+) {
+  const ghostAddresses = (existingUserInstance?.Addresses?.filter(
+    ({ ghost_address }: AddressAttributes) => !!ghost_address,
+  ) || []) as AddressAttributes[];
+
+  for (const ghost of ghostAddresses) {
+    const replacementAddress = addressInstances.find(
+      ({ community_id, ghost_address }) =>
+        !ghost_address && community_id === ghost.community_id,
+    );
+
+    // should always exist, but check for it to avoid null check error
+    if (replacementAddress) {
```

Comment:
We may want to return some data if the address was successfully replaced, or not, so we can surface some kind of message saying "migrated posts from discourse!"

Reply:
We should probably create a new ticket for this since it also requires UI work.

Reply:
@jnaviask after responding all your comments, this is the one that might require a follow-up ticket in coordination with the product team